### ISSUE TITLE
filter strategies

### ISSUE THREAD

**Issue by biozzq** [ANGSD/angsd#59]
Hi @ANGSD 

After running `SnpFilters` for each population, I want to get the reliable SNP sites, so I want to filter all the sites using following cutoff for each chromosome, I will remove the following sites,
`Individuals information < 90%`
`hwe_pvalue < 1e-3`
`baseQ_pvalue < 1e-5`
`SB3_fisher_score < 90 quantile `

for minor allele frequency, I will keep following sites,
`snp_pvalue < 1e-6 and maf > 1/(2*Individual.Number) ` or 
`snp_pvalue == 1 and maf > 1/(2*Individual.Number) when the major allele is same to ref allele` or 
`snp_pvalue == 1  but the major allele is different from ref allele`

But I find the cutoff for `SB3_fisher_score`  is `1`, so I suspect that I should count the duplicate `SB3_fisher_score` only once and then calculate the 90 quantile, because the most of the `SB3_fisher_score`  is `1` in the snpStat file.  Is this a proper way for defining the cutoff of `SB3_fisher_score`?

Many thanks.
Best

**Comment by biozzq** [comment]
Hi @ANGSD

I have six populations including one wild population and five domesticated populations. I use all these populations to do `SnpFilter` with following command for each chromosome,
`angsd -bam bam.list -only_proper_pairs 1 -uniqueOnly 1 -remove_bads 1 -minQ 20 -minMapQ 30 -C 50 -ref ref.fa -nThreads 30 -r 1 -out 1 -doMaf 1 -doMajorMinor 1 -GL 1 -setMinDepth 800 -setMaxDepth 8000 -doCounts 1 -P 5 -dosnpstat 1`
but when I use above strategies for all populations, I find it just retains a little SNP sites (about 10% when compared with GATK pipeline (also exclude the low quality sites)), a large proportion of sites are excluded by maf and snp_pvalue. I think I had made some mistakes on above `SnpFilter` command or the filter strategies, and I have no ideas how to figure them out for these days.  So I hope you can  give me a hand. 

Many thanks

**Comment by biozzq** [comment]
Hi @ANGSD,

I really need your help, this problem have troubled me for several days. Many thanks.

Best

**Comment by ANGSD** [comment]
Hi, If you dont supply any parameter to -dosnpstat 1, it won't do any
filtering.There are different ways to calculate strandbias for ngs data,
which is the reason why there are 3 different stats in angsd. I would look
at the distribution of these.
The 3 sb stats are defined in this paper: on guo 2012 BMC genomics 2013
13:666

**Comment by biozzq** [comment]
Hi @ANGSD ,

Yes, I can supply without any parameter to `-dosnpstat 1`, but what I specified parameters are all about the quality of mapping and the global depth generated from `-doDepth 1`.  In order to generate the true positive SNPs sites, I first filter the false positive sites using above cutoff (mainly about p_value cutoff), but I just get a small part sites when compared with genotypes called by GATK refer to its best practices. Maybe I misunderstood `angsd commands` when I just want to get the reliable SNP sites. 

I know these three different measures in `angsd` and lower values mean less strand bias and higher scores mean a more severe strand bias. The author considered SNPs with strand bias score in the top 10% as extreme. So I used 10% as the SB3 cutoff, but I just doubt whether I should count the duplicate `SB3_fisher_score` only once and then calculate the 90 quantile.

many thanks.
